InsiteTeamTask/Repositories/DataRepository.cs

```diff
@@ -17,47 +15,136 @@ public List<Attendance> GetAttendanceListFor(int gameNumber)
+                return filteredProduct.GameId == gameNumber && filteredProduct.SeasonId == seasonNumber;
+            })
+                .Select(member =>
                 {
-                    Barcode = "N/A",
-                    MemberId = members[i].Id
+                    return new Attendance
+                    {
+                        Barcode = "N/A",
+                        MemberId = member.Id
+                    };
                 });
-            }
 
-            for (int i = 0; i < tickets.Count; i++)
-            {
-                attendanceList.Add(new Attendance()
+            attendanceList.AddRange(filteredMembers);
+
+            var filteredTickets = tickets.Where(ticket => {
+                var filteredProduct = products.Single(product => product.Id == ticket.ProductId);
+
+                return filteredProduct.GameId == gameNumber && filteredProduct.SeasonId == seasonNumber;
+            })
+                .Select(ticket =>
                 {
-                    Barcode = tickets[i].Barcode,
-                    MemberId = 0
+                    return new Attendance
+                    {
+                        Barcode = ticket.Barcode,
+                        MemberId = 0
+                    };
                 });
-            }
+
+            attendanceList.AddRange(filteredTickets);
 
             return attendanceList;
         }
 
-        public List<Season> GetSeasons(int eventId)
+        public List<Attendance> GetAttendanceListFor(string productCode)
         {
-            throw new NotImplementedException();
+            MockDataService service = new MockDataService();
+
+            List<Member> members = service.Members().ToList();
+            List<Ticket> tickets = service.Tickets().ToList();
+
+            List<Attendance> attendanceList = new List<Attendance>();
+
+            var filteredMembers = members
+                .Where(member => member.ProductId == productCode)
+                .Select(member => new Attendance
+                {
+                    Barcode = "N/A",
+                    MemberId = member.Id
+                });
+
+            attendanceList.AddRange(filteredMembers);
+
+            var filteredTickets = tickets
+                .Where(ticket => ticket.ProductId == productCode)
+                .Select(ticket => new Attendance
+                {
+                    Barcode = ticket.Barcode,
+                    MemberId = 0
+                });
+
+            attendanceList.AddRange(filteredTickets);
+
+            return attendanceList;
         }
 
-        public List<Event> GetEvents()
+        // I have removed the 2 methods that were here GetSeasons(EventId) and GetEvents() as they had no implementaions and there are no fields defined for Events so I couldn't link a season to an event or vice versa.
+
+        // The below methods were created to allow me to load the initial data needed to populate the dropdowns
+        public List<Season> GetSeasons()
```

Comment:
If you changed the return type of these methods to `IEnumerable<Season>`, you'd be able to pass the result from the `IDataService` straight through to the calling function rather than casting it to a list here.